src/dune_sexp/syntax.ml

```diff
@@ -250,15 +250,28 @@ module Key = struct
 end
 
 module Error_msg = struct
+  let lang_or_using name = if name = "dune" then "lang" else "using"
+
   let since t ver ~what =
-    let lang_or_using = if t.name = "dune" then "lang" else "using" in
     Printf.sprintf
       "%s is only available since version %s of %s. Please update your dune-project file \
        to have (%s %s %s)."
       what
       (Version.to_string ver)
       t.desc
-      lang_or_using
+      (lang_or_using t.name)
+      t.name
+      (Version.to_string ver)
+  ;;
+
+  let since_config t ver ~what =
```

Comment:
Is there a point in having this be two separate functions? To me it seems like what it does is

```ocaml
sprintf "%s is only available since version %s of %s. Please update your %s file to have (%s %s %s)."

i.e. the file to be mentioned can also just be an argument.
```

Reply:
Yea! Everything here can be super simplified if we take the file name as an argument. When deciding how to fix this my initial thought was to alter the `since` function as follows:
```ocaml
let since ?(file = "dune-project") ?(fatal = true) ?what t ver = ...
```
Due to the fact that `since` is used all over the place I felt that it might be worth proposing and implementation which preserved the original signature and also provided a way to control the entire error message. If you think that altering the signature as shown above would be okay I'd be happy to move towards this type of representation, this would also have the added benefit simplifying/resolving some of the other changes you mention below!

Reply:
Your correct! It's functionally the same thing just with a different string describing a different file. I can condense this into a single function.